### ISSUE TITLE
A file is reporting a "referenced without its extension" warning, but the text of the warning is not in the reported file

### ISSUE THREAD

**Issue by ryanpeach** [agent-ecosystem/skill-validator#85]
## What happened?

A failing CI build https://github.com/ryanpeach/skills/actions/runs/30497824823/job/90730631865?pr=7

```markdown
⚠ file references/lint.yml is referenced without its extension (as references/lint in references/check-skill-nonlinks.md) — include the .yml extension so agents can reliably locate the file
```

My understanding: The file `references/check-skill-nonlinks.md` contains the text `references/lint` and the linter wants me to change it to `references/lint.yml`

The relevant file: https://github.com/ryanpeach/skills/blob/d9e56f808baa194632da83b7fa4684151cb90e77/plugins/skill-linting/skills/skill-linting/references/check-skill-nonlinks.md

The problem: This file does not include the text "references/lint" at all.

## What did you expect?

No warning, or warning with the proper information

## How to reproduce, Environment, Additional context

CI is linked

https://github.com/ryanpeach/skills/actions/runs/30497824823/job/90730631865?pr=7
